### PR TITLE
Share doctree between builders

### LINKED ISSUE #4363: Is it possible to control the "force" resp. "-E" option for Sphinx? (turns out it's actually the "-d" option)
My builds take quite a bit of time and memory because I'm using a custom Sphinx source parser that executes Jupyter notebooks (http://nbsphinx.readthedocs.io/).

I've seen that the `-E` option is used when calling Sphinx, which means that for every output format (HTML, JSON, PDF, ...) the source files are parsed (and in my case the Jupyter notebooks are executed) again and again.

Parsing the source files multiple times doesn't (and shouldn't!) change anything, it just burns some of my (and your) precious build time and memory.

Is it possible to disable this on a per-project basis?

I guess there is some reason to do this in the first place, but if not, it should probably be deactivated globally?

I've looked through the source code and found that this is controlled by the `force` argument to the builder class and just about everywhere the default is `force=False`.
But it seems that the actual build process is somehow started with `force=True`.

### PR DESCRIPTION
**Pull request by stsewd**
We are reducing build time with this.
And also saving a little of space in the builders.

According to sphinx, we are safe doing this:

> the doctrees can be shared between all builders

http://www.sphinx-doc.org/en/master/man/sphinx-build.html#cmdoption-sphinx-build-d

Closes #4363

### CODE REVIEW

**Comment by stsewd** [comment]
Tested this in the local instance inside docker, I don't see much improvements in the time like it did outside (https://github.com/rtfd/readthedocs.org/issues/4363#issuecomment-432847628)

But the obvious improvement is the size.

**Comment by agjohnson** [comment]
Perhaps this would make sense under a feature flag. I feel like none of us remember why we had to separate the doctree in the first place, I'm guessing we'll find out really quick :slightly_smiling_face:

File: readthedocs/projects/models.py [hunk 1]
```diff
@@ -1313,31 +1313,40 @@ def add_features(sender, **kwargs):
...
+    SHARE_SPHINX_DOCTREE = 'share_sphinx_doctree'
 
     FEATURES = (
         (USE_SPHINX_LATEST, _('Use latest version of Sphinx')),
         (USE_SETUPTOOLS_LATEST, _('Use latest version of setuptools')),
         (ALLOW_DEPRECATED_WEBHOOKS, _('Allow deprecated webhook views')),
         (PIP_ALWAYS_UPGRADE, _('Always run pip install --upgrade')),
-        (SKIP_SUBMODULES, _('Skip git submodule checkout')), (
+        (SKIP_SUBMODULES, _('Skip git submodule checkout')),
+        (
```
Comment (stsewd) [inline readthedocs/projects/models.py]: Sorry but it was bothering me seeing this _js_ kind of indentation p:

**Comment by stsewd** [comment]
@agjohnson feature flag added

**Comment by humitos** [comment]
BTW, shouldn't we update this docs https://docs.readthedocs.io/en/latest/guides/feature-flags.html ?

**Comment by stsewd** [comment]
> BTW, shouldn't we update this docs https://docs.readthedocs.io/en/latest/guides/feature-flags.html ?

I don't think so, we are just testing this to then remove it, and this shouldn't affect users at all if what we are doing is ok. So, feature flag will be removed soon if things go well :)

**Comment by stsewd** [comment]
I'm adding a trello card to keep track of adding this flag to some projects

**Comment by ericholscher** [comment]
Pretty sure this is going to blow up in production. We can have multiple projects running on the same builder across different versions, which will cause weird race conditions from this I'm 99% sure.

**Comment by stsewd** [comment]
But the `_build/` folder is created for each version, we can only have race conditions if we trigger a build to the same version twice, the same race condition would be seen before this change.

**Comment by ericholscher** [comment]
Ah right, it must have been another issue then. Guess we'll see in prod as Anthony said.

**Pull request merged**